docs/source/python/memory.rst

```diff
@@ -277,6 +277,95 @@ types than with normal Python file objects.
    !rm example.dat
    !rm example2.dat
 
+Efficiently Writing and Reading Arrow Arrays
```

Comment:
It seems this should go into `ipc.rst`. `memory.rst` is about the low-level memory and IO APIs.

Reply:
I'm open to suggestions about where to best put this. I picked `memory.rst` because the title is "Memory and IO Interfaces" and it covers input streams, memory mapped files and reading memory mapped data.

Reply:
Sorry for the delay. Basically, reading and writing Arrow arrays uses the IPC layer, so I do think it would be better in `ipc.rst`. `memory.rst` is only dealing with raw bytes data.

Reply:
moved to `ipc.rst` as suggested